core/src/main/java/org/elasticsearch/action/admin/cluster/health/TransportClusterHealthAction.java

```diff
@@ -191,6 +194,9 @@ private boolean prepareResponse(final ClusterHealthRequest request, final Cluste
         if (request.waitForStatus() != null && response.getStatus().value() <= request.waitForStatus().value()) {
             waitForCounter++;
         }
+        if (request.waitForQuorumActive() == true && response.isQuorumActive() == true) {
+            waitForCounter++;
+        }
```

Comment:
unrelated but can we add an assertion at the bottom of this method to ensure `waitForCounter <= waitFor`